### PR TITLE
Support custom description separator in full_description

### PR DESCRIPTION
**Pull request by chikamichi**
Hi,

Here's a little piece of Quality Of Life improvement.

Hopefully both the feature and its implementation get a positive review :) As far as a feature spec goes, I wasn't sure about where to add it, so I'd like for the reviewer(s) to make a suggestion.

### CODE REVIEW

File: lib/rspec/core/configuration.rb [hunk 1]
```diff
@@ -268,13 +268,18 @@ def exclude_pattern=(value)
...
+      # Run all examples if none match the configured filters.
       # (default: `false`).
       # @deprecated Use {#filter_run_when_matching} instead for the specific
       #   filters that you want to be ignored if none match.
       add_setting :run_all_when_everything_filtered
 
       # @macro add_setting
+      # Define a custom separator to join nested examples/contexts' descriptions.
+      # (default: " ")
+      add_setting :description_separator
```
Comment (chikamichi) [inline lib/rspec/core/configuration.rb]: I'm wondering whether I should define a custom setter to `.freeze` the custom separator.

**Comment by xaviershay** [comment]
Thank you for the suggestion! What would you use this for? What's a typical value?

We're generally pretty reluctant to add surface area to our config, since it's already pretty wide, so we have a pretty high bar for new configuration settings we accept. And then, we generally want to see config succeed in a plugin first before pulling into core.

I'm closing for now, but feel free to re-open with further justification!

**Comment by chikamichi** [comment]
Hi,

Well, the way RSpec displays nested examples by default is just not that great. The space separator makes it very hard to read through the output. From my experience, on many non-trivial code bases, one will have multiple nested meta-examples (`describe`, `context`) and it comes in handy "separating" them visually in the "Failures:" summary.

Instead of an option, I had considered proposing changing the default separator from `" "` to `", "` but that seemed a little bold.

What do you think about the default separator, what has been your experience with nested contexts? I know of several people who agree with me but then, that's not a good case for the feature :)

**Comment by chikamichi** [comment]
> we generally want to see config succeed in a plugin first before pulling into core.

Now I've been quick at submitting a PR and didn't consider coming up with a plugin. It sounds like a nice solution, so I'll explore that meanwhile you hopefully get back at me with more insights :) Thank you!

**Comment by xaviershay** [comment]
Sorry I haven't replied yet, will take a look in the next couple of days!

**Comment by xaviershay** [comment]
Interesting, this hadn't occurred to me as a problem before, probably because I match on line numbers rather than spec names.

Definitely sounds like a good candidate for a plugin!

**Comment by chikamichi** [comment]
Hi, I totally missed your feedback from 7 months ago! Anyway, haven't had time for this but it still is a nice idea that might happen some day...

**Pull request closed without merging**